common/lib/user.js

```diff
@@ -1,25 +1,63 @@
-function User (token) {
-  const permissions = []
-  const authorities = token.authorities || []
-
-  if (authorities.includes('ROLE_PECS_POLICE')) {
-    permissions.push(
-      ...[
-        'moves:view:by_location',
-        'moves:download:by_location',
-        'move:view',
-        'move:create',
-        'move:cancel',
-      ]
-    )
-  }
-
-  if (authorities.includes('ROLE_PECS_SUPPLIER')) {
-    permissions.push(...['moves:view:all', 'moves:download:all'])
-  }
-
-  this.userName = token.user_name
-  this.permissions = permissions
+const Sentry = require('@sentry/node')
+
+const referenceDataService = require('../../common/services/reference-data')
+// TODO: Remove once we can determine locations from auth
+const { CURRENT_LOCATION_UUID } = require('../../config')
+
+function User () {}
+
+User.prototype = {
+  init (token) {
```

Comment:
I'm not sure instantiation with an access token is the best way to separate concerns. Also the structure of the access token content can differ quite a lot between providers. I'd prefer we could instantiate using the user data (name, roles, locations) as named arguments. This would probably improve the clarity of the tests as well.

Reply:
I think I see what you're saying.

So should the User class be simpler in that its not concerned with retrieving the full location object from the API (whether that be by ID like now, or by agency code in the future)?

The main purpose of this change is so that our user object has the concept of `locations` that it can access. We can then check parts of the app like viewing moves for a specific location or changing the current default location against this list.

I didn't fully like that this User class became async to be honest, made it a bit trickier to test.

Maybe we should use the reference data service within the auth middleware then and then pass that list of locations in the user object?

Reply:
I agree.